lib/xnvme_nvm.c

```diff
@@ -112,6 +112,21 @@ xnvme_prep_nvm(struct xnvme_cmd_ctx *ctx, uint8_t opcode, uint32_t nsid, uint64_
 	ctx->cmd.nvm.nlb = nlb;
 }
 
+int
+xnvme_nvm_dsm(struct xnvme_cmd_ctx *ctx, uint32_t nsid, uint8_t nr, bool ad, bool idw, bool idr,
+	      struct xnvme_spec_dsm_range *dsm_range)
+{
+	ctx->cmd.common.opcode = XNVME_SPEC_NVM_OPC_DATASET_MANAGEMENT;
+	ctx->cmd.common.nsid = nsid;
+	ctx->cmd.dsm.nr = nr;
+	ctx->cmd.dsm.ad = ad;
+	ctx->cmd.dsm.idw = idw;
+	ctx->cmd.dsm.idr = idr;
+
+	return xnvme_cmd_pass(ctx, NULL, 0, (void *)dsm_range,
+			      sizeof(struct xnvme_spec_dsm_range) * nr);
```

Comment:
This should should be passed to the ``dbuf_nbytes`` argument of ``xnvme_cmd_pass()`` not ``mbuf_nbytes``.

Reply:
Additionally, then:
``sizeof(struct xnvme_spec_dsm_range) * nr``
Could be expressed as:
``sizeof(*dsm_range) * nr``
This is convenient as established as relation between the argument and its type.

Reply:
Fixed!